### PR TITLE
Refactor the code in order to work with latest siddhi released version and few improvements

### PR DESCRIPTION
**Pull request by AnuGayan**
## Purpose
By this,
1. Siddhi 4.1.x will be supported
2. Now SIDDHISDK_HOME can be set as an environment variable and use jar in that location
3. Supports MacOS
4. Improved README.md

### CODE REVIEW

**Review by sajithshn — APPROVED** [review]

File: Tests/SiddhiCoreTests/Resources/Extensions4/pom.xml [hunk 1]
```diff
@@ -1,4 +1,22 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c)  2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (ksdperera) [inline Tests/SiddhiCoreTests/Resources/Extensions4/pom.xml:3]: Incorrect licenses year, it should be 2018

File: __PySiddhi4Proxy/threadfix_c_code/makefile [hunk 2]
```diff
@@ -1,18 +1,41 @@
+# ---------------------------------------------------------------------------
+#  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (ksdperera) [inline __PySiddhi4Proxy/threadfix_c_code/makefile:2]: Please fix the License header year in all the places

File: Tests/SiddhiCoreTests/Resources/Extensions4/pom.xml (hunk 1 above)
Comment (AnuGayan) [reply Tests/SiddhiCoreTests/Resources/Extensions4/pom.xml:3]: This file was added on 2017. That's why I added 2017 as the license year

File: __PySiddhi4Proxy/threadfix_c_code/makefile (hunk 2 above)
Comment (AnuGayan) [reply __PySiddhi4Proxy/threadfix_c_code/makefile:2]: This file was added in 2017. That's why I added 2017 as the license year

**Pull request merged**